src/ProcessorPluginManager.php

```diff
@@ -10,48 +10,39 @@
 namespace Zend\Log;
 
 use Zend\ServiceManager\AbstractPluginManager;
+use Zend\ServiceManager\Exception\InvalidServiceException;
+use Zend\ServiceManager\Factory\InvokableFactory;
 
 /**
  * Plugin manager for log processors.
  */
 class ProcessorPluginManager extends AbstractPluginManager
 {
-    /**
-     * Default set of processors
-     *
-     * @var array
-     */
-    protected $invokableClasses = [
-        'backtrace'      => 'Zend\Log\Processor\Backtrace',
-        'psrplaceholder' => 'Zend\Log\Processor\PsrPlaceholder',
-        'referenceid'    => 'Zend\Log\Processor\ReferenceId',
-        'requestid'      => 'Zend\Log\Processor\RequestId',
+    protected $aliases = [
+        'backtrace'      => Processor\Backtrace::class,
+        'psrplaceholder' => Processor\PsrPlaceHolder::class,
+        'referenceid'    => Processor\ReferenceId::class,
+        'requestid'      => Processor\RequestId::class
     ];
 
-    /**
-     * Allow many processors of the same type
-     *
-     * @var bool
-     */
-    protected $shareByDefault = false;
+    protected $factories = [
+        Processor\Backtrace::class      => InvokableFactory::class,
+        Processor\PsrPlaceHolder::class => InvokableFactory::class,
+        Processor\ReferenceId::class    => InvokableFactory::class,
+        Processor\RequestId::class      => InvokableFactory::class
+    ];
 
     /**
-     * Validate the plugin
-     *
-     * Checks that the processor loaded is an instance of Processor\ProcessorInterface.
-     *
-     * @param  mixed $plugin
-     * @return void
-     * @throws Exception\InvalidArgumentException if invalid
+     * {@inheritdoc}
      */
-    public function validatePlugin($plugin)
+    public function validate($plugin)
```

Comment:
As noted on another plugin manager in here, no need to override the `validate` method; just define the `$instanceOf` property to the value `Processor\ProcessorInterface::class`.